server/market/market.go

```diff
@@ -325,6 +323,29 @@ func (m *Market) OrderFeed() <-chan *updateSignal {
 	return bookUpdates
 }
 
+func (m *Market) FeedDone(feed <-chan *updateSignal) bool {
```

Comment:
If there is a need for the book router to shutdown prior to the market stopping, this will prevent the market from hanging.  It is also used in tests to change the feed channel, although I think that could be eliminated.